### ISSUE TITLE
FTBFS with sshagent

### ISSUE THREAD

**Issue by the-nic** [keepassxreboot/keepassxc#1238]
When trying to build KeePassXC with the option `WITH_XC_SSHAGENT=ON` I get a compile error:
```
[13/143] Building CXX object src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o
FAILED: src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o 
/usr/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_EXCEPTIONS -DQT_STRICT_ITERATORS -DQT_WIDGETS_LIB -DWITH_APP_BUNDLE -I../src -Isrc -I../src/zxcvbn -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -D_FORTIFY_SOURCE=2 -Werror=format-security -Wcast-align -pie -fPIE -std=c++11 -O2 -g -DNDEBUG   -fPIC -std=gnu++11 -MMD -MT src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o -MF src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o.d -o src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o -c ../src/sshagent/AgentSettingsPage.cpp
In file included from ../src/sshagent/AgentSettingsPage.cpp:20:0:
../src/sshagent/AgentSettingsWidget.h:24:36: fatal error: ui_AgentSettingsWidget.h: No such file or directory
compilation terminated.
[14/143] Building CXX object src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o
FAILED: src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o 
/usr/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_EXCEPTIONS -DQT_STRICT_ITERATORS -DQT_WIDGETS_LIB -DWITH_APP_BUNDLE -I../src -Isrc -I../src/zxcvbn -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -D_FORTIFY_SOURCE=2 -Werror=format-security -Wcast-align -pie -fPIE -std=c++11 -O2 -g -DNDEBUG   -fPIC -std=gnu++11 -MMD -MT src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o -MF src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o.d -o src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o -c ../src/sshagent/AgentSettingsWidget.cpp
In file included from ../src/sshagent/AgentSettingsWidget.cpp:19:0:
../src/sshagent/AgentSettingsWidget.h:24:36: fatal error: ui_AgentSettingsWidget.h: No such file or directory
compilation terminated.
[15/143] Building CXX object src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o
FAILED: src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o 
/usr/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_EXCEPTIONS -DQT_STRICT_ITERATORS -DQT_WIDGETS_LIB -DWITH_APP_BUNDLE -I../src -Isrc -I../src/zxcvbn -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -D_FORTIFY_SOURCE=2 -Werror=format-security -Wcast-align -pie -fPIE -std=c++11 -O2 -g -DNDEBUG   -fPIC -std=gnu++11 -MMD -MT src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o -MF src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o.d -o src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o -c src/sshagent/sshagent_automoc.cpp
In file included from src/sshagent/moc_AgentSettingsWidget.cpp:9:0,
                 from src/sshagent/sshagent_automoc.cpp:2:
src/sshagent/../../../src/sshagent/AgentSettingsWidget.h:24:36: fatal error: ui_AgentSettingsWidget.h: No such file or directory
```

## Expected Behavior

No compile error

## Current Behavior

see above 

## Possible Solution

Patching `src/sshagent/AgentSettingsWidget.h` helps:
Replace `#include "ui_AgentSettingsWidget.h"` with `#include "sshagent/ui_AgentSettingsWidget.h"`

## Steps to Reproduce (for bugs)

1. Use KDE neon 
2. Use out of tree build
3. Build project (ninja/make is irrelevant)

Operating system: 
Description:    KDE neon User Edition 5.11
Release:        16.04
Codename:       xenial

CPU architecture: x64
Kernel: Linux nic-x250 4.10.0-40-generic
GCC: 5.4.0

**Comment by hifi** [comment]
Reproduced on the same distro. Need some help figuring out what the correct fix is as other `ui_` includes work fine in other directories.

**Comment by yan12125** [comment]
Might be related to CMake version? CMake 3.8 changes paths for uic-generated files. Another example: https://github.com/lxde/lxqt-config/pull/118

**Comment by the-nic** [comment]
I have CMake 3.5.1, so in this case only newer versions work?

**Comment by droidmonkey** [comment]
AUTOUIC has been a headache for too long, i might revert that feature and go back to explicitly naming UIC files.

**Comment by yan12125** [comment]
This issue also breaks recent TeamCity builds. For example:
```
[07:32:18]In file included from /opt/buildagent/work/7ce463a40a422d79/src/sshagent/AgentSettingsWidget.cpp:19:
[07:32:18]/opt/buildagent/work/7ce463a40a422d79/src/sshagent/AgentSettingsWidget.h:24:10: fatal error: 'ui_AgentSettingsWidget.h' file not found
[07:32:18]#include "ui_AgentSettingsWidget.h"
[07:32:18]         ^
[07:32:18]In file included from /opt/buildagent/work/7ce463a40a422d79/build/src/sshagent/sshagent_automoc.cpp:2:
[07:32:18]In file included from /opt/buildagent/work/7ce463a40a422d79/build/src/sshagent/moc_AgentSettingsWidget.cpp:9:
[07:32:18]/opt/buildagent/work/7ce463a40a422d79/build/src/sshagent/../../../src/sshagent/AgentSettingsWidget.h:24:10: fatal error: 'ui_AgentSettingsWidget.h' file not found
[07:32:18]#include "ui_AgentSettingsWidget.h"
[07:32:18]         ^
[07:32:18]1 error generated.
[07:32:18]make[2]: *** [src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o] Error 1
[07:32:18]make[2]: *** Waiting for unfinished jobs....
[07:32:18]In file included from /opt/buildagent/work/7ce463a40a422d79/src/sshagent/AgentSettingsPage.cpp:20:
[07:32:18]/opt/buildagent/work/7ce463a40a422d79/src/sshagent/AgentSettingsWidget.h:24:10: fatal error: 'ui_AgentSettingsWidget.h' file not found
[07:32:18]#include "ui_AgentSettingsWidget.h"
[07:32:18]         ^
[07:32:19]1 error generated.
[07:32:19]make[2]: *** [src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o] Error 1
[07:32:19]1 error generated.
[07:32:19]make[2]: *** [src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o] Error 1
[07:32:20]make[1]: *** [src/sshagent/CMakeFiles/sshagent.dir/all] Error 2
[07:32:20]make[1]: *** Waiting for unfinished jobs....
[07:32:22]make: *** [all] Error 2
[07:32:23]Process exited with code 2
```
(From https://ci.keepassxc.org/viewLog.html?buildId=3348&tab=buildResultsDiv&buildTypeId=KeepassXC_DebugClang)

**Comment by phoerious** [comment]
Yes, I've noticed that, but can't fix it before this evening.
